delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/auth/AuthenticateAs.scala

```diff
@@ -0,0 +1,12 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.auth
+
+import ch.epfl.bluebrain.nexus.delta.kernel.Secret
+import pureconfig.ConfigReader
+import pureconfig.generic.semiauto.deriveReader
+
+
+case class AuthenticateAs(user: String, password: Secret[String], realm: String)
```

Comment:
This needs to be an ADT: Anonymous is the other alternative

Reply:
Rename to `Credentials` ?

We also need one with Token at least for a while.
For retrocompatibility reasons but also because it will be a safety net if we detect a bug in the new implementation.
The remote storage is a sensitive thing :)

Reply:
Also the realm should be a label

Reply:
Can you give an example of where we have an ADT in the config? Is it correct that we need to have a type field which states which class a user is specifying?